engine/src/main/java/com/google/android/fhir/db/impl/dao/SyncedResourceDao.kt

```diff
@@ -28,6 +29,11 @@ interface SyncedResourceDao {
 
   @Insert(onConflict = OnConflictStrategy.REPLACE) suspend fun insert(entity: SyncedResourceEntity)
 
+  @Transaction
+  suspend fun insertAll(resources: List<SyncedResourceEntity>) {
+    resources.forEach { resource -> insert(resource) }
+  }
```

Comment:
```suggestion
  @Insert(onConflict = OnConflictStrategy.REPLACE)
  suspend fun insertAll(resources: List<SyncedResourceEntity>)
```

you don't need to do it manually, room can receive a list to insert.

Reply:
done